src/alfasim_sdk/_internal/hook_specs.py

```diff
@@ -178,6 +178,64 @@ def update_plugins_secondary_variables(ctx: "void*") -> "int":
     """
 
 
+def update_plugins_secondary_variables_time_explicit(ctx: "void*") -> "int":
+    """
+    **c++ signature** : ``HOOK_UPDATE_PLUGINS_SECONDARY_VARIABLES_TIME_EXPLICIT(void* ctx)``
+
+    Internal simulator hook to update plugin's secondary variables time explicit.
+    This is called only once at the end of ALFAsim's update internal secondary variables workflow.
+
+    :param ctx: ALFAsim's plugins context
+    :returns: Return OK if successful or anything different if failed
+
+    Example of usage:
+
+    .. code-block:: c++
+        :linenos:
+        :emphasize-lines: 1
+
+        HOOK_UPDATE_PLUGINS_SECONDARY_VARIABLES_TIME_EXPLICIT(ctx)
+        {
+            int errcode = -1;
+            int size_U = -1;
+            int size_E = -1;
+            int liq_id = -1;
+            errcode = alfasim_sdk_api.get_field_id(
+                ctx, &oil_id, "oil");
+            double* vel;
+            VariableScope Fields_OnFaces = {
+                GridScope::FACE,
+                MultiFieldDescriptionScope::FIELD,
+                TimestepScope::CURRENT
+            }
+            errcode = alfasim_sdk_api.get_simulation_array(
+                ctx, &vel, (char*) "U", Fields_OnFaces, liq_id, &size_U);
+            double* kinetic_energy;
+            char* name = "kinetic_energy_of_oil";
+            int global_idx = 0;
+            errcode = alfasim_sdk_api.get_plugin_variable(
+                ctx,
+                (void**) (&kinetic_energy),
+                name,
+                global_idx,
+                TimestepScope::CURRENT,
+                &size_E);
+            if (size_U != size_E){
+                return OUT_OF_BOUNDS;
+            }
+            for (int i =0; i < size_U; ++i){
+                kinetic_energy[i] = vel[i] * vel[i] / 2.;
+            }
+            return OK;
+        }
+
+    In the example above the variable ``kinetic_energy_of_oil`` was registered as a global variable, but its value is
+    obtained for `oil field` and only at the end of the time steps' linear solver (explicit approach). If this
+    variable would be calculated to all fields then the ``global_idx`` would be substituted by ``field_idx`` and it
+    would be performed to each `field`.
```

Comment:
How does the value is obtained for the oil_field if the variable is global and the `oil_id` variable is not used?

Reply:
Putz
I made this example based on the one used at `update_plugins_secondary_variables`, but I think that line [221](https://github.com/ESSS/alfasim-sdk/pull/366/files/990d9b4b09f0cd1088f46847624c7142d02f3a36#diff-57379b1558ae80d93105233310befbcc92749b75f1add1edbe7c699a0b2b3700R211) should be using `oil_id` instead of `liq_id` and the text is saying that we are only computing the kinetic_energy of oil field because we are using only de oil velocity 🤔

Reply:
Nice, thanks for correcting it.